File: school-management-backend/src/main/java/com/school/management/service/AttendanceService.java
package com.school.management.service;

import com.school.management.dto.AttendancePercentageResponse;
import com.school.management.dto.AttendanceRequest;
import com.school.management.entity.Attendance;

import java.time.LocalDate;
import java.util.List;

public interface AttendanceService {
    List<Attendance> getAllAttendance();
    Attendance markAttendance(AttendanceRequest request);
    List<Attendance> markBatchAttendance(List<AttendanceRequest> requests);
    List<Attendance> getAttendanceByDate(LocalDate date);
    List<Attendance> getAttendanceByStudentId(Long studentId);
    List<Attendance> getAttendanceByClassAndDate(Long classId, LocalDate date);
    AttendancePercentageResponse getAttendancePercentage(Long studentId);
}

File: school-management-frontend/src/services/attendanceService.js
import api from '../api/axios';

const attendanceService = {
  getAllAttendance: () => api.get('/attendance'),
  markAttendance: (data) => api.post('/attendance', data),
  markBatchAttendance: (dataList) => api.post('/attendance/batch', dataList),
  getAttendanceByDate: (date) => api.get(`/attendance/date/${date}`),
  getAttendanceByStudent: (studentId) => api.get(`/attendance/student/${studentId}`),
  getAttendanceByClassAndDate: (classId, date) => api.get(`/attendance/class/${classId}/date/${date}`),
  getAttendancePercentage: (studentId) => api.get(`/attendance/student/${studentId}/percentage`),
};

export default attendanceService;
